packages/ceramic-core/src/document.ts

```diff
@@ -440,7 +493,7 @@ class Document extends EventEmitter {
   }
 
   get owners (): string[] {
```

Comment:
Looks like this should be `string[] | undefined` if the state or metadata object is undefined.

Reply:
Metadata should never be undefined, neither should `owners`.

Reply:
@oed 👍

Reply:
There's a mismatch between the type annotations and the code then, the `?` shouldn't be necessary in `this._doctype.state?.metadata?.owners` if these objects can never be undefined.

Reply:
Hm, afaik the `state` should never be undefined either? I might be missing something in the implementation details though?

Reply:
nope, `state` should be always defined. Fixed it 👍